### PR TITLE
SAK-42557 - Add group info to message detail view

### PR DESCRIPTION
**Pull request by adrianmticarum**
Commit to add the rol and group information of the message author to the message detail view.
Several visual issues in the messages list page fixed. Minor clean and refactoring of the code.

### CODE REVIEW

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java [hunk 1]
```diff
@@ -20,53 +20,16 @@
...
-import org.sakaiproject.api.app.messageforums.MessageForumsMessageManager;
-import org.sakaiproject.api.app.messageforums.MessageForumsTypeManager;
-import org.sakaiproject.api.app.messageforums.PrivateForum;
-import org.sakaiproject.api.app.messageforums.PrivateMessage;
-import org.sakaiproject.api.app.messageforums.PrivateMessageRecipient;
-import org.sakaiproject.api.app.messageforums.PrivateTopic;
-import org.sakaiproject.api.app.messageforums.SynopticMsgcntrManager;
-import org.sakaiproject.api.app.messageforums.Topic;
-import org.sakaiproject.api.app.messageforums.UserPreferencesManager;
+import org.sakaiproject.api.app.messageforums.*;
```
Comment (mpellicer) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: I'm not a big fan of using * for imports, I don't want to block the PR for this but let's see what others think

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java [hunk 2]
```diff
@@ -108,9 +71,18 @@
...
+import javax.faces.application.FacesMessage;
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+import javax.faces.event.ValueChangeEvent;
+import javax.faces.model.SelectItem;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
```
Comment (mpellicer) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: I'm not a big fan of using * for imports, I don't want to block the PR for this but let's see what others think

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java (hunk 1 above)
Comment (adrianfish) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: I agree with Miguel. Leave the explicit imports in and just allow IDEs to collapse them. That's the common pattern in Sakai.

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java [hunk 3]
```diff
@@ -4577,4 +4519,21 @@ private LRS_Statement getStatementForUserSentPvtMsg(String subject, SAKAI_VERB s
     public String getAttachmentReadableSize(final String attachmentSize) {
       return FileUtils.byteCountToDisplaySize(Long.parseLong(attachmentSize));
     }
+
+    public String getRolAndGroupsOfSender() {
```
[the remark this answers is not in the document]
Comment (adrianmticarum) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: Of course. Done!

File: msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsgDetail.jsp [hunk 4]
```diff
@@ -123,12 +123,20 @@
...
-                            <h:outputText value=" #{msgs.pvt_closeb}" /> 
+                            <h:outputText value=" #{msgs.pvt_closeb}" />
+                        </td>
+                    </tr>
+                    <tr>
+                        <th>
+                            <h:outputText value="#{msgs.pvt_group}"/>
+                        </th>
+                        <td>
+                            <h:outputText value="#{PrivateMessagesTool.rolAndGroupsOfSender} "/>
```
[the remark this answers is not in the document]
Comment (adrianmticarum) [reply msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsgDetail.jsp]: Done!

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java (hunk 1 above)
Comment (adrianmticarum) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: Ok, I'll keep this in mind. Done!

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java (hunk 2 above)
Comment (adrianmticarum) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: Done!

**Comment by mpellicer** [comment]
Thanks @adrianmticarum

**Pull request merged**